### PR TITLE
Add splunk to neo4j

### PR DESCRIPTION
**Pull request by adampalay**
Configuration Pull Request
---

I'm closing https://github.com/edx/configuration/pull/3943 in favor of this PR. It does the same things, but also:

1. Increases the maximum amount of open files to 40000, as suggested by [neo4j](https://neo4j.com/developer/kb/how-do-i-set-max-open-files-for-debian-installs/)
2. Updates the version of java that neo4j uses to run

I've run this play against a sandbox to verify that it completes and that the logs are getting outputted where I expect them to be outputted. However, I don't know how to verify that, once splunk forwarding is enabled, they will get forwarded to splunk.

@feanil , could you please take a look at this PR too? Thanks!

Make sure that the following steps are done before merging:

  - [x] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [x] Add an entry to the CHANGELOG.
  - [x] Have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?

### CODE REVIEW

File: playbooks/roles/neo4j/defaults/main.yml [hunk 1]
```diff
@@ -21,12 +21,14 @@ NEO4J_SERVER_NAME: "localhost"
...
 neo4j_apt_repository: "deb http://debian.neo4j.org/repo stable/"
 neo4j_version: "3.0.3"
+neo4j_defaults_file: "/etc/default/neo4j"
 neo4j_server_config_file: "/etc/neo4j/neo4j.conf"
 neo4j_wrapper_config_file: "/etc/neo4j/neo4j-wrapper.conf"
 neo4j_https_port: 7473  # default in package is 7473
 neo4j_http_port: 7474  # default in package is 7474
 neo4j_listen_address: "0.0.0.0"
 neo4j_heap_max_size: "3000"
+neo4j_log_dir: "{{ COMMON_LOG_DIR }}/neo4j"
```
Comment (feanil) [inline playbooks/roles/neo4j/defaults/main.yml]: I don't think we need to do this.  It's fine to leave this file in '/var/log'

File: playbooks/roles/splunkforwarder/defaults/main.yml [hunk 2]
```diff
@@ -76,6 +76,10 @@ SPLUNKFORWARDER_LOG_ITEMS:
     recursive: true
     index: '{{ COMMON_ENVIRONMENT }}-{{ COMMON_DEPLOYMENT }}'
     sourcetype: 'rabbitmq'
+  - source: '{{ COMMON_LOG_DIR }}/neo4j'
```
Comment (feanil) [inline playbooks/roles/splunkforwarder/defaults/main.yml]: Set this to '/var/log/neo4j'

**Comment by adampalay** [comment]
@feanil responded to your comments!

File: playbooks/edx-east/neo4j.yml [hunk 3]
```diff
@@ -4,6 +4,7 @@
   gather_facts: True
   vars:
     CLUSTER_NAME: 'coursegraph'
+    ENABLE_SPLUNKFORWARDER: False
```
Comment (feanil) [inline playbooks/edx-east/neo4j.yml]: `COMMON_ENABLE_SPLUNKFORWARDER`?

**Review by feanil — COMMENTED** [review]
@adampalay one more question, then I think this is good to go.

File: playbooks/edx-east/neo4j.yml (hunk 3 above)
Comment (adampalay) [reply playbooks/edx-east/neo4j.yml]: I actually don't know. I'm following suit what the [other roles do](https://github.com/edx/configuration/search?utf8=%E2%9C%93&q=ENABLE_SPLUNKFORWARDER%3A+False&type=) in setting this to `False` and running their `splunkforwarder` role when `COMMON_ENABLE_SPLUNKFORWARDER` is set.

I assume `COMMON_ENABLE_SPLUNKFORWARDER` is set elsewhere?

Comment (feanil) [reply playbooks/edx-east/neo4j.yml]: @adampalay I think this line can just be removed,  I think the var changed but no one removed the old var... I'll put in a separate ticket to remove that var from the other playbooks.

Comment (adampalay) [reply playbooks/edx-east/neo4j.yml]: Roger that!

**Comment by adampalay** [comment]
@feanil , I removed that var!

**Review by feanil — APPROVED** [review]

**Pull request merged**